bus-mapping/src/evm/opcodes/calldatacopy.rs

```diff
@@ -99,9 +90,21 @@ fn gen_copy_event(
 ) -> Result<CopyEvent, Error> {
     let rw_counter_start = state.block_ctx.rwc;
 
-    let memory_offset = geth_step.stack.nth_last(0)?.low_u64();
+    let memory_offset = geth_step.stack.nth_last(0)?;
     let data_offset = geth_step.stack.nth_last(1)?;
-    let length = geth_step.stack.nth_last(2)?.as_u64();
+    let length = geth_step.stack.nth_last(2)?;
+
+    let call_ctx = state.call_ctx_mut()?;
+    let memory = &mut call_ctx.memory;
+    memory.extend_for_range(memory_offset, length);
+
+    let memory_updated = {
```

Comment:
current memory update doesn't come to `state.call_ctx_mut()?.memory` because of clone happens in L102.  which will cause evm circuit&copy circuit fail.

Reply:
good refactor style!.  cargo test found multiple copy circuit& evm circuit failures(I think most due to ditto comments), suggest to fix the test first to avoid import new failures to `memory_opt` branch, then good to merge.

Reply:
The memory update is managed now by the function `memory_write_word`, no longer by writing into the `memory` buffer.

Reply:
will test it again and I can help fix if still have test failures.